plugins/scaffolder-backend-module-gitlab/src/actions/editGitlabIssueAction.ts

```diff
@@ -0,0 +1,249 @@
+    })
+    .optional(),
+  removeLabels: z
+    .string({
+      description: 'Comma-separated label names to remove from an issue.',
+    })
+    .optional(),
+  stateEvent: z
+    .nativeEnum(IssueStateEvent, {
+      description:
+        'The state event of an issue. To close the issue, use close, and to reopen it, use reopen.',
+    })
+    .optional(),
+  title: z.string().describe('The title of an issue.').optional(),
+  updatedAt: z
+    .string()
+    .describe(
+      'When the issue was updated. Date time string, ISO 8601 formatted',
+    )
+    .regex(
+      /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(?:\.\d{3})?Z$/,
+      'Invalid date format. Use YYYY-MM-DDTHH:mm:ssZ or YYYY-MM-DDTHH:mm:ss.SSSZ',
+    )
+    .optional(),
+  weight: z
+    .number({ description: 'The issue weight' })
+    .min(0, 'Valid values should be equal or greater than zero')
+    .max(10, 'Valid values should be equal or less than 10')
+    .optional(),
+});
+
+const editIssueOutputProperties = z.object({
+  issueUrl: z.string({ description: 'Issue WebUrl' }),
+  projectId: z.number({
+    description: 'The project id the issue belongs to WebUrl',
+  }),
+  issueId: z.number({ description: 'The issues Id' }),
+  issueIid: z.number({
+    description: "The issues internal ID of a project's issue",
+  }),
+  state: z.string({ description: 'The state event of an issue' }),
+  title: z.string({ description: 'The title of an issue.' }),
+  updatedAt: z.string({ description: 'The last updated time of the issue.' }),
+});
+
+/**
+ * Creates a `gitlab:issues:edit` Scaffolder action.
+ *
+ * @param options - Templating configuration.
+ * @public
+ */
+export const editGitlabIssueAction = (options: {
+  integrations: ScmIntegrationRegistry;
+}) => {
+  const { integrations } = options;
+  return createTemplateAction({
+    id: 'gitlab:issues:edit',
+    description: 'Edit a Gitlab issue.',
+    examples,
+    schema: {
+      input: commonGitlabConfig.merge(editIssueInputProperties),
+      output: editIssueOutputProperties,
+    },
+    async handler(ctx) {
+      try {
+        const {
+          repoUrl,
+          projectId,
+          title,
+          addLabels,
+          removeLabels,
+          issueIid,
+          description,
+          confidential = false,
+          assignees = [],
+          updatedAt = '',
+          dueDate,
+          discussionLocked = false,
+          epicId,
+          labels,
+          issueType,
+          milestoneId,
+          stateEvent,
+          weight,
+          token,
+        } = commonGitlabConfig.merge(editIssueInputProperties).parse(ctx.input);
+
+        const { host } = parseRepoUrl(repoUrl, integrations);
+        const api = getClient({ host, integrations, token });
+
+        let isEpicScoped = false;
+
+        if (epicId) {
+          isEpicScoped = await checkEpicScope(
+            api as any as InstanceType<typeof Gitlab>,
```

Comment:
```suggestion
            api,
```

You can also remove this casting from `createGitlabIssueAction` too, it's not necessary.

Reply:
Happy to make the chance, it was a copy of the `CreateGitlabIssueAction.ts` happy to make the change to both if it is what we wish to do.